pkg/converter/patcher.go

```diff
@@ -23,3 +25,43 @@ func addAnnotations(annotations *map[string]string, addAnnotations map[string]st
 		(*annotations)[k] = v
 	}
 }
+
+func isModifiedImage(image string, modifiedImages []string) bool {
+	for _, modifiedImage := range modifiedImages {
+		// check if we have an exact match
+		if image == modifiedImage {
+			return true
+		}
+		// check if we have a suffix match. For this we must ensure that the modifiedImage string starts with a "/",
+		// otherwise we might get nonsensical partial matches
+		if !strings.HasPrefix(modifiedImage, "/") {
+			modifiedImage = "/" + modifiedImage
+		}
+		if strings.HasSuffix(image, modifiedImage) {
+			return true
+		}
+	}
+	return false
+}
+
+func PatchDeployments(deployments []apps.Deployment, modifiedImages []string, forceRestartAnnotation map[string]string) {
+	// don't use 'range', getting a pointer to an array element does not work with 'range'
+	for i := 0; i < len(deployments); i++ {
+		for _, container := range deployments[i].Spec.Template.Spec.Containers {
+			if isModifiedImage(container.Image, modifiedImages) {
+				addAnnotations(&deployments[i].Spec.Template.Annotations, forceRestartAnnotation)
```

Comment:
We _could_ `break` here and move to the next deployment. Same below

Reply:
Thanks, fixed.